### PR TITLE
containerinfra: add a few missing labels fields

### LINKED ISSUE #2376: Missing fields in containerinfra towards label supports
There are several fields related to labels in containerinfra:
- The merge_label option in nodegroups (the support has been added in clusters but not in nodegroups)
-  The fields "labels_added", "labels_overridden" and "labels_skipped" in both nodegroups and clusters in the result

### PR DESCRIPTION
**Pull request by MrFreezeex**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

Fixes #2376

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/magnum/blob/20413d0c45dc7a6426072bd889bfb71c73fa62b1/magnum/api/controllers/v1/nodegroup.py#L129-L145
https://github.com/openstack/magnum/blob/20413d0c45dc7a6426072bd889bfb71c73fa62b1/magnum/api/controllers/v1/cluster.py#L187-L203

### CODE REVIEW

**Comment by mandre** [comment]
Thanks for the PR. You should normally have received a nice welcoming message for your first PR but our automation failed. I'm copying the content below:

> Thank you for submitting your first PR! Be sure that we will be looking at it but keep in mind
this sometimes takes a while.
Please let the maintainers know if your PR has not got enough attention after a few days.
If any doubt, please consult our PR [tutorial](https://github.com/gophercloud/gophercloud/blob/master/docs/contributor-tutorial/step-05-pull-requests.md). 

I'll review your PR in a minute.

**Comment by mandre** [comment]
This LGTM, however we don't have CI exercising your changes against a real openstack. We have yet to port the Magnum CI to github actions - I'd like we do that before merging your PR.

**Comment by mandre** [comment]
Hey @MrFreezeex, we've recently merged the change to enable CI for Magnum in #2379. Could you rebase this PR on top of master so that we see your change in action before we merge it?

**Comment by MrFreezeex** [comment]
> Hey @MrFreezeex, we've recently merged the change to enable CI for Magnum in #2379. Could you rebase this PR on top of master so that we see your change in action before we merge it?

Thanks for the great work, I just did that :D

**Review by mandre — APPROVED** [review]
Meh, this was pointless - we're still not exercising this new code in the acceptance tests. I couldn't get Magnum to deploy a cluster in CI and had to skip the tests that would have exercised this code :-/
Sorry for all the trouble, let's merge your PR as it looks good and the risk it breaks anything is rather low.

**Pull request merged**
